lib/options.js

```diff
@@ -0,0 +1,136 @@
+"use strict";
+
+var path = require("path");
+var merge = require("lodash.merge");
+var deprecate = require("./util/deprecate");
+
+function Options(eyeglass, options, sassArg) {
+  options = getSassOptions.apply(null, arguments);
+
+  Object.keys(options).forEach(function(key) {
+    this[key] = options[key];
+  }.bind(this));
+}
+
+function eyeglassOptionsFromNodeSassArg(arg) {
+  // if it looks like node-sass...
+  if (arg && arg.render && arg.renderSync && arg.info) {
+    // throw a deprecation warning
+    deprecate([
+      "You should no longer pass `sass` directly to Eyeglass. Instead pass it as an option:",
+      "var options = eyeglass(sassOptions, {",
+      "  engines: {",
+      "    sass: require('node-sass')",
+      "  }",
+      "});"
+    ].join("\n  "));
+
+    // and convert it the correct format
+    return {
+      engines: {
+        sass: arg
+      }
+    };
+  }
+}
+
+function includePathsFromEnv() {
+  if (process.env.SASS_PATH) {
+    return process.env.SASS_PATH.split(path.delimiter).map(function(dir) {
+      return path.resolve(process.cwd(), dir);
+    });
+  }
+}
+
+function migrateEyeglassOptionsFromSassOptions(sassOptions, eyeglassOptions) {
+  // migrates the following properties from sassOptions to eyeglassOptions
+  [
+    "root",
+    "cacheDir",
+    "buildDir",
+    "httpRoot",
+    "assetsHttpPrefix",
+    "assetsRelativeTo",
+    "strictModuleVersions"
+  ].forEach(function(option) {
+    if (eyeglassOptions[option] === undefined && sassOptions[option] !== undefined) {
+      deprecate([
+        "`" + option + "` should be passed into the eyeglass options rather than the sass options:",
+        "var options = eyeglass(\/* sassOptions *\/, {",
+        "  " + option + ": ...",
+        "});"
+      ].join("\n  "));
+
+      eyeglassOptions[option] = sassOptions[option];
+      delete sassOptions[option];
+    }
+  });
+}
+
+function defaultSassOptions(options) {
+  options.includePaths = defaultValue(options.includePaths, includePathsFromEnv() || []);
+  return options;
+}
+
+function defaultEyeglassOptions(options, eyeglass) {
+  // default root dir
+  options.root = defaultValue(options.root, process.cwd());
+  // default cache dir
+  options.cacheDir = defaultValue(options.cacheDir, path.join(options.root, ".eyeglass_cache"));
+  // default engines
+  options.engines = defaultValue(options.engines, {});
+  options.engines.sass = defaultValue(options.engines.sass, require("node-sass"));
+  options.engines.eyeglass = defaultValue(options.engines.eyeglass, eyeglass);
```

Comment:
yo dawg, optception!

I thought we decided eyeglass didn't need to be in the options.